sdk/batch/Microsoft.Azure.Batch/src/Common/RetryPolicyCommon.cs

```diff
@@ -23,6 +23,7 @@ internal static bool ShouldRetry(Exception exception, OperationContext operation
                         && statusCode != 429 // Too many reqeuests
                         )
                         || statusCode == 501 // Not Implemented
+                        || statusCode == 503 // ServiceUnavailable
```

Comment:
Don't we want to retry 503 errors? 501 errors are different because it likely indicates a permanent failure.